### ISSUE TITLE
GlosSiTarget fails to close when exiting from the game

### ISSUE THREAD

**Issue by Yuzuriha-San** [LemmusLemmus/GlosSI-Integration-Playnite#35]
Please note that when exiting via the steam BPM overlay and not from the game itself, GlosSiTarget process is killed without problem. This only happens when closing from inside the game or emulator.

![6vjXNKhlCY](https://github.com/user-attachments/assets/375cd4b2-43a3-4870-a7f1-9a517e436f79)

Playnite shows that the game is no longer running but Steam still shows it as running and it will kill GlosSiTarget if I press STOP manually.

![steamwebhelper_FSR64x0h19](https://github.com/user-attachments/assets/0c5fe8e1-ef53-4cbf-aa4a-9a0a537d10c2)

So why is this a problem? If GlosSiTarget fails to close and you go on to open the same game or another integrated game, Playnite will freeze and stop responding until you kill the previous GlosSiTarget process. Then it will unfreeze and open said game with the overlay. That means if you use Playnite Fullscreen mode overlay and browse a game and try to launch it from there. GlosSiTarget will fail to close and Playnite will hang until you kill GlosSiTarget manually from Task Manager or Steam.

My current Windows version is 24h2 LTSC. This wasn't an issue when I was on 23h2 Iot Enterprise.

**Comment by Yuzuriha-San** [comment]
So as I initially thought, this behavior happens only on 24h2. I went back to 23h2 due to numerous bugs and performance issues and this issue was resolved.

[I suggest anyone wanting to upgrade to 24h2 not to do so.](https://www.neowin.net/reviews/windows-11-24h2-vs-23h2-in-place-upgrade-performance-in-2025-some-super-surprising-results/)

**Comment by mbutera11** [comment]
@OdinIsLord Did this ever come back for you? I have this problem as well, and I'm using 23H2.

**Comment by Yuzuriha-San** [comment]
> [@OdinIsLord](https://github.com/OdinIsLord) Did this ever come back for you? I have this problem as well, and I'm using 23H2.

Yes, Idk if it was Microsoft porting 24h2 features to 23h2 via the monthly channel updates. I honestly gave up on it.

But this is not the extensions fault as this also occurs with the standalone GlosSi program, It seems to that the latest GlosSi version is force creating a global overlay of some sort on top of the individual one for each game.

@LemmusLemmus might be able to comment on it.

**Comment by Yuzuriha-San** [comment]
@mbutera11 Okay, the issue was indeed with the GlosSi program, I was using the latest GoW Ragnarok fix snapshot version provided by the dev. I was confused because he took down all prior versions and left only that up on Github and took his site down. Ended up finding the 0.1.2.0 stable version and I can confirm after installing it this issue is resolved, GlosSi closes the overlay without any problem and kills the whole process every single time.
